app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt

```diff
@@ -0,0 +1,126 @@
+package com.github.HumanLearning2021.HumanLearningApp
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.view.View
+import android.widget.Button
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.LifecycleOwner
+import com.google.common.util.concurrent.ListenableFuture
+
+/*
+Activity where an administrator can take a picture to add it to a selected data set
+ */
+class AddPictureActivity : AppCompatActivity() {
+
+    private lateinit var cameraProviderFuture: ListenableFuture<ProcessCameraProvider>
+    private lateinit var requestPermissionLauncher: ActivityResultLauncher<String>
+    private lateinit var openLabelSelectionLayoutListener: View.OnClickListener
+
+    private val dummyLabels = arrayOf("label1", "label2", "label3", "label4", "label5", "label6", "label7", "label8", "label9", "label10")
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setupRequestPermissionLauncher()
+        setupOpenLabelSelectionLayoutListener()
```

Comment:
I think it's not ideal to make your code depend on state. If later you need to modify something and you accidentally use requestPermissionLauncher before the call to setupRequestPermissionLauncher, it will not work. Maybe the simpler solution would be to make the function return a reference to the newly created object. It also allows you to remove the class attributes.

Reply:
Agreed